File: SimpleAsyncTask/app/src/main/java/com/example/gustavogobetti/simpleasynctask/MainActivity.java
package com.example.gustavogobetti.simpleasynctask;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    // The TextView where we will show results
    private TextView mTextView;

    //Key for saving the state of the TextView
    private static final String TEXT_STATE = "currentText";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mTextView = findViewById(R.id.textView1);

        //Restore TextView if there is a savedInstanceState
        if(savedInstanceState != null){
            mTextView.setText(savedInstanceState.getString(TEXT_STATE));
        }

    }


    public void startTask(View view) {
        //Put the message in the TextView
        mTextView.setText(R.string.napping);
        //Start the Async
        new SImpleAsyncTask(mTextView).execute();
    }


    protected void onSaveInstancesState(Bundle outState) {
        super.onSaveInstanceState(outState);
        // Save the state of the TextView
        outState.putString(TEXT_STATE, mTextView.getText().toString());

    }
}

File: spinnerA2/app/src/main/java/com/example/gustavogobetti/spinnera2/MainActivity.kt
package com.example.gustavogobetti.spinnera2

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.Spinner
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {



    internal lateinit var sp : Spinner

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        sp = findViewById(R.id.spinnerON) as Spinner
        val movies = arrayOf("Catch me if you can","Fight Club"
            ,"Forrest Gump" , "Good Will Hunting","Pulp Fiction","The GodFather",
            "The Hangover", "The Shaw Shank Redemption",
            "Titanic")
        val adapter = ArrayAdapter(this,android.R.layout.simple_list_item_1,movies)
        sp.adapter = adapter


    fun Spinner.setOnItemSelectedListener(listener: AdapterView.OnItemSelectedListener){
        onItemSelectedListener = listener
    }

//    sp.setOnItemSelectedListener = object :AdapterView.OnItemSelectedListener {
//        override fun onItemSelected(parent : AdapterView<*>?, view: View?, position: Int, id: Long){
//
//        }
//
//        override fun onNothingSelected(parent: AdapterView<*>?){
//
//        }
//    }

}
}
